lake/index/writer.go

```diff
@@ -109,36 +110,64 @@ type indexer struct {
 	wg      sync.WaitGroup
 }
 
-func newIndexer(ctx context.Context, u iosrc.URI, def *Definition, r zbuf.Reader) (*indexer, error) {
+func newIndexer(ctx context.Context, path iosrc.URI, ref Reference, r zbuf.Reader) (*indexer, error) {
```

Comment:
`Reference` is big. Use a pointer.
```suggestion
func newIndexer(ctx context.Context, path iosrc.URI, ref *Reference, r zbuf.Reader) (*indexer, error) {
```